kagami/routes/admin/resource.py

```diff
@@ -0,0 +1,113 @@
+
+
+from fastapi import APIRouter, HTTPException
+from pydantic import BaseModel, HttpUrl, field_validator
+from typing import Literal
+
+from kagami.routes.deps import SupervisorDeps
+
+resource_router = APIRouter(prefix="/admin/resource")
+
+class ResourceConfig(BaseModel):
+    name: str
+    upstream_url: HttpUrl
+    provider_method: Literal["rsync", "git"]
+    retry: bool = True
+    
+    @field_validator('name')
+    def name_isvalid(self, v):
+        if not v.isalnum():
+            raise ValueError('name must be alphanumeric')
+        return v.lower()
+    
+    class Config:
+        json_schema_extra = {
+            "example": {
+                "name": "ubuntu",
+                "upstream_url": "https://mirrors.ubuntu.com/",
+                "provider_method": "rsync",
+                "retry": True
+            }
+        }
+
+@resource_router.get("/list")
+async def list_resources(supervisor: SupervisorDeps):
+    """get all resources"""
+    resources = await supervisor.list_resource()
+    return [
+        {
+            "name": name,
+            "status": resource.status.value,
+            "providers_count": len(resource.providers)
+        }
+        for name, resource in resources.items()
+    ]
+
+@resource_router.get("/{resource_name}/detail")
+async def get_resource_detail(resource_name: str, supervisor: SupervisorDeps):
+    """get resource detail"""
+    resource = supervisor.resources.get(resource_name)
+    if not resource:
+        raise HTTPException(status_code=404, detail="Resource not found")
+    
+    return {
+        "name": resource_name,
+        "status": resource.status.value,
+        "providers": [
+            {
+                "id": p.replica_id,
+                "worker": p.worker_addr,
+                "status": p.provider_status.value
+            }
+            for p in resource.providers
+        ]
+    }
+
+@resource_router.post("/{resource_name}/sync")
+async def sync_resource(resource_name: str, supervisor: SupervisorDeps):
+    """Sync resource"""
+    resource = supervisor.resources.get(resource_name)
+    if not resource:
+        raise HTTPException(status_code=404, detail="Resource not found")
+    
+
+    return {"message": "Sync started"}
+
+@resource_router.post("/create")
+async def create_resource(config: ResourceConfig, supervisor: SupervisorDeps):
+    """Create new resource"""
+    try:
+        # Pydantic自动验证输入数据
+        resource = await supervisor.create_resource(
+            name=config.name,
+            upstream_url=config.upstream_url,
+            provider_method=config.provider_method,
+            retry=config.retry
```

Comment:
We don't have create_resource in our api definition on apifox. Please carefully read apifox documentation. ResourceInfo is just a data collection of providers and the maintaining of ResourceInfo is automatically.